Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,208 @@
+
+### Monitoring ViewController lifecycle directly instead of relying on `ScreenLifecycleEvent`
+
+With the introduction of `ScreenLifecycleEvent`, developers should no longer manually monitor the viewController's lifecycle events. Instead, this kind of monitoring can now easily be done via the `send(_: ScreenLifecycleEvent)` method.
+
+Here's a brief example of how it was typically done in both Builders and FlowControllers:
+
+```swift
+viewController
+            .reactive
+            .signal(for: #selector(UIViewController.viewWillDisappear))
+            .discardValues()
+            .observeValues(dismiss)
+```
+
+Nowadays, this can be done much more elegantly (thanks Anders!):
+
+```swift
+func send(_ event: ScreenLifecycleEvent) {
+        switch (event, state.status) {
+        case (.didAppear, .showing):
+            runSomething()
+        default:
+            break
+        }
+    }
+```
+
+As of June 7th, 2019, we still have 16 results in 13 different files.
+
+In order to detect this, we could simply monitor occurences of the following regex:
+
+`\.signal\(for: \#selector\(UIViewController\.`
+
+We'd have to restrict this rule to Builders and FlowControllers only, therefore we'd have to whitelist filenames that matched `.*Builder.swift` and `.*FlowController.swift`.
+
+This rule could be enforced with either tool but I'd recommend using SwiftLint for this since due to the low number of warnings it would trigger in the current state of our database. Also, I believe we should try to use Danger for particularly important warnings in order not to spam the pull request with comments.
+
+
+### Deprecated calls of DesignLibrary components (aka pre-NVL components)
+
+Given that NVL is scheduled for release within two weeks, using NVL components is pivotal to Babylon's overall look and feel. With the exception of the empty space component (even though it has its own NVL equivalent), every other pre-NVL component used inside Renderer should trigger a warning. Nevertheless, this would require that everyone relied exclusively on NVL components and while I am not entirely sure this is the case right now the future belongs to NVL.
+
+The following regex currently yields 1333 results in 216 files: `Component\.(?!EmptySpace)\b`.
+
+The sheer number of warnings immediately excludes SwiftLint, which leaves us with:
+a) marking these methods as deprecated
+b) using Danger to raise a comment whenever a PR is raised
+
+In this particular case I believe we could use both approaches actually.
+
+### Detect Current usage outside Builders et al
+
+`Current` is a very powerful concept whose usage is only allowed in a very restrict number of places. As such, we should enable automatic monitoring in order to prevent it from being abused throughout our codebase. Nevertheless, Current is still being used in a lot of places (598 results across 144 files) and we should go through every case individually before commiting to a decision:
```

Comment:
```suggestion
`Current` is a very powerful concept whose usage is only allowed in a very restrict number of places. 
Check [Current Guide](./Technical-Documents/Current-guide.md) . As such, we should enable automatic monitoring in order to prevent it from being abused throughout our codebase. Nevertheless, Current is still being used in a lot of places (598 results across 144 files) and we should go through every case individually before commiting to a decision:
```